core/broker/server.go

```diff
@@ -57,8 +54,8 @@ func (b *brokerRPC) Associate(stream pb.Broker_AssociateServer) error {
 		if err != nil {
 			return err
 		}
-		if !uplink.Validate() {
-			return grpcErrf(codes.InvalidArgument, "Invalid Uplink")
+		if err := uplink.Validate(); err != nil {
+			return errors.BuildGRPCError(errors.NewErrInvalidArgument("Uplink", err.Error()))
```

Comment:
Be consistent with `errors.Wrap` vs wrapping it in `errors.NewErrInvalidArgument`